### PR TITLE
Task5 Patterns

### PR DESCRIPTION
**Pull request by BorysFe**

### CODE REVIEW

File: src/main/java/com/trello/BoardPage.java [hunk 1]
```diff
@@ -112,26 +123,26 @@ public void closeBoard() {
...
 
     public void deleteBoardPermanently() {
         closeBoard();
-        waitUtils.waitVisibilityOfElementShort(deleteBoard);
+        waitUtils.waitVisibilityOfElementLong(deleteBoard);
         deleteBoard.click();
         waitUtils.waitVisibilityOfElementShort(confirmationDeleteBoardBtn);
         confirmationDeleteBoardBtn.click();
-        waitUtils.waitInvisibilityOfElementShort(confirmationDeleteBoardBtn);
+
```
Comment (edward-astapov) [inline src/main/java/com/trello/BoardPage.java]: Remove extra line

File: src/main/java/com/trello/BoardPage.java [hunk 2]
```diff
@@ -112,26 +123,26 @@ public void closeBoard() {
...
+        waitUtils.waitVisibilityOfElementLong(textAfterDeleteBoard);
     }
 
     public String getSearchFieldAttribute(String attributeName) {
         waitUtils.waitVisibilityOfElementShort(searchField);
         return searchField.getAttribute(attributeName);
     }
 
-    public String getBoardTitleAttribute(String attributeName) {
+    public String getBoardTitleAttribute() {
```
Comment (edward-astapov) [inline src/main/java/com/trello/BoardPage.java]: Current method behavior does not match title

File: src/main/java/com/trello/LogInPage.java [hunk 3]
```diff
@@ -29,35 +37,59 @@ public LogInPage(WebDriver driver) {
...
+            passwordField.sendKeys(userPassword);
+            waitUtils.waitElementToBeClickableShort(logInBtnGreen);
+            logInBtnGreen.click();
+        }
     }
 
-    public void logInWithEmail(String userEmail) {
-        waitUtils.waitElementToBeClickableShort(openLogInPageButton);
-        openLogInPageButton.click();
+    public void logInNewUserFactory(User user) {
```
Comment (edward-astapov) [inline src/main/java/com/trello/LogInPage.java]: logInNewUser

File: src/main/java/com/trello/WaitUtils.java [hunk 4]
```diff
@@ -122,4 +122,13 @@ public void waitElementToBeClickableShort(WebElement element) {
     public void waitElementToBeClickableLong(WebElement element) {
         waitElementToBeClickable(element, longTimeout);
     }
+
+    public boolean waitVisibilityOfElementBooleanShort(WebElement element) {
```
Comment (edward-astapov) [inline src/main/java/com/trello/WaitUtils.java]: isElementVisibleShortTimeout (all boolean methods should named like a question)

File: src/test/java/com/trello/authorization/AuthorisationTest.java [hunk 5]
```diff
@@ -17,27 +17,34 @@
...
     }
 
     @BeforeMethod
     public void openEnvironment() {
+        driver = new ChromeDriver();
         driver.get("https://trello.com");
+        logInPage = new LogInPage(driver);
+        boardPage = new BoardPage(driver);
+        logInPage.openLoginPageFactory(driver);
+
```
Comment (edward-astapov) [inline src/test/java/com/trello/authorization/AuthorisationTest.java]: Remove extra line

File: src/test/java/com/trello/authorization/AuthorisationTest.java [hunk 6]
```diff
@@ -17,27 +17,34 @@
...
     }
 
     @AfterMethod
     public void signOffFromTrello() {
-        BoardPage boardPage = new BoardPage(driver);
         boardPage.logOutIfAuthorised();
+        driver.close();
```
Comment (edward-astapov) [inline src/test/java/com/trello/authorization/AuthorisationTest.java]: .quit()
Update in all tests classes

**Review by edward-astapov — CHANGES_REQUESTED** [review]
Need works

File: src/test/java/com/trello/authorization/AuthorisationTest.java (hunk 5 above)
Comment (BorysFe) [reply src/test/java/com/trello/authorization/AuthorisationTest.java]: Done

File: src/main/java/com/trello/WaitUtils.java (hunk 4 above)
Comment (BorysFe) [reply src/main/java/com/trello/WaitUtils.java]: Done

File: src/main/java/com/trello/BoardPage.java (hunk 2 above)
Comment (BorysFe) [reply src/main/java/com/trello/BoardPage.java]: Done

File: src/main/java/com/trello/BoardPage.java (hunk 1 above)
Comment (BorysFe) [reply src/main/java/com/trello/BoardPage.java]: Done

File: src/main/java/com/trello/LogInPage.java (hunk 3 above)
Comment (BorysFe) [reply src/main/java/com/trello/LogInPage.java]: Done

File: src/test/java/com/trello/authorization/AuthorisationTest.java (hunk 6 above)
Comment (BorysFe) [reply src/test/java/com/trello/authorization/AuthorisationTest.java]: In these changes - I will delete the annotation @AfterClass (browser quit())

**Review by edward-astapov — APPROVED** [review]

**Pull request merged**